packages/mobile/src/store/init/setupCrypto/CryptoDelegator.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  CryptoServicePayload,
+  CryptoServiceResponse,
+  SocketActionTypes
+} from '@quiet/state-manager'
+import { Socket } from 'socket.io-client'
+import { NodeEnv } from '../../../utils/const/NodeEnv.enum'
+
+export interface ArrayBufferViewWithPromise extends ArrayBufferView {
+  _promise?: Promise<ArrayBufferView>
+}
+
+const SUBTLE_METHODS = [
+  'encrypt',
+  'decrypt',
+  'sign',
+  'verify',
+  'digest',
+  'generateKey',
+  'deriveKey',
+  'deriveBits',
+  'importKey',
+  'exportKey',
+  'wrapKey',
+  'unwrapKey'
+]
+
+export class CryptoDelegator {
+  socket: Socket
+
+  calls: {
+    [id: string]: {
+      resolvePromise: (value: any) => void
+    }
+  }
+
+  constructor(socket: Socket) {
+    this.socket = socket
+  }
+
+  public get subtle(): SubtleCrypto {
+    const subtle = {}
```

Comment:
I think we should memoize this object.
As it is rigth now I think you create a new object every time this getter is called